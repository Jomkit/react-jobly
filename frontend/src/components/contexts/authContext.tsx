import { createContext } from 'react';
import { authInterface } from '../../types';

export const authContext = createContext<authInterface>({} as authInterface);