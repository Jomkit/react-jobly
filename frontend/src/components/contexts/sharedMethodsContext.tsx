import { createContext } from 'react';
import { sharedMethodsInterface } from '../../types';

export const sharedMethodsContext = createContext<sharedMethodsInterface>({} as sharedMethodsInterface);