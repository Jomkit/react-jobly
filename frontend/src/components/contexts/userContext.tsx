import { createContext } from "react";
import { userInterface } from "../../types";

export const userContext = createContext<userInterface | null>(null);