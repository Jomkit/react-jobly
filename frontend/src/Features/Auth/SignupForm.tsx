import { useContext } from 'react'
import UserForm from './UserForm';
import { sharedMethodsContext } from '../../components/contexts/sharedMethodsContext';

const SignupForm = () => {
    const { signup } = useContext(sharedMethodsContext);
  return (
    <>
        <UserForm pageTitle="Sign Up!" handleSubmit={signup} />
    </>
  )
}

export default SignupForm