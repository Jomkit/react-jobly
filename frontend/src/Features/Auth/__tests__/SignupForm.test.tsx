import { render } from "@testing-library/react"
import userEvent from "@testing-library/user-event";
import { MemoryRouter } from "react-router-dom";
import { sharedMethodsContext } from "../../../components/contexts/sharedMethodsContext";
import SignupForm from "../SignupForm";
import React from "react"
import { TESTUSER2 } from "../../../__tests__/_testCommon";

function setup(tsx: React.ReactNode) {
    return {
        user: userEvent.setup(),
        ...render(tsx)
    }
}

const signup = vi.fn();
const login = vi.fn();
const logout = vi.fn();

const TestSignupForm = () => {
    return (
        <MemoryRouter initialEntries={["/"]}>
            <sharedMethodsContext.Provider value={{signup, login, logout}} >
                <SignupForm />
            </sharedMethodsContext.Provider>
        </MemoryRouter>
    )
}

it('should render without crashing', () => { 
    render(<TestSignupForm />)
})

it('should match snapshot', () => { 
    const { asFragment } = render(<TestSignupForm />);
    expect(asFragment()).toMatchSnapshot();
})

it('should update name field on change', async () => { 
    const { getByLabelText, user } = setup(<TestSignupForm />);

    const nameInput = getByLabelText('Username');
    await user.type(nameInput, 'foo');
    expect(nameInput).toHaveValue('foo');
})

it('should redirect to home page on successful submit', async () => { 
    const { getByLabelText, getByText, user } = setup(<TestSignupForm />);

    const userNameInput = getByLabelText('Username');
    const passwordInput = getByLabelText('Password');
    const firstNameInput = getByLabelText('First Name');
    const lastNameInput = getByLabelText('Last Name');
    const emailInput = getByLabelText('Email');
    const submitBtn = getByText('Submit');

    await user.type(userNameInput, TESTUSER2.username);
    await user.type(passwordInput, TESTUSER2.password);
    await user.type(firstNameInput, TESTUSER2.firstName);
    await user.type(lastNameInput, TESTUSER2.lastName);
    await user.type(emailInput, TESTUSER2.email);
    
    await user.click(submitBtn);

    expect(signup).toHaveBeenCalled();
    expect(signup).toHaveBeenCalledWith(TESTUSER2);
})