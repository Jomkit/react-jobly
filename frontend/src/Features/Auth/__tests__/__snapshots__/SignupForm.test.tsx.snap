// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <h1
    class="mt-4 text-white"
  >
    Sign Up!
  </h1>
  <form
    action="#"
    class="d-flex flex-column col col-6 m-auto p-5 pt-3 border border-3 rounded border-dark bg-white"
  >
    <label
      for="username"
    >
      Username
    </label>
    <input
      class="text-input"
      id="username"
      name="username"
      type="text"
      value=""
    />
    <label
      for="password"
    >
      Password
    </label>
    <input
      class="text-input"
      id="password"
      name="password"
      type="password"
      value=""
    />
    <label
      for="firstName"
    >
      First Name
    </label>
    <input
      class="text-input"
      id="firstName"
      name="firstName"
      type="text"
      value=""
    />
    <label
      for="lastName"
    >
      Last Name
    </label>
    <input
      class="text-input"
      id="lastName"
      name="lastName"
      type="text"
      value=""
    />
    <label
      for="email"
    >
      Email
    </label>
    <input
      class="text-input"
      id="email"
      name="email"
      type="email"
      value=""
    />
    <button
      class="btn btn-primary mt-2"
      type="submit"
    >
      Submit
    </button>
  </form>
</DocumentFragment>
`;
